include/pybind11/pytypes.h

```diff
@@ -1829,7 +1829,7 @@ class capsule : public object {
             // guard if destructor called while err indicator is set
             error_scope error_guard;
             auto destructor = reinterpret_cast<void (*)(void *)>(PyCapsule_GetContext(o));
```

Comment:
This line was introduced with PR #752:

https://github.com/pybind/pybind11/pull/752/files

@wjakob Why is this `PyCapsule_GetContext()` and not `PyCapsule_GetDestructor()`? 

I see it's self-consistent within pybind11, `PyCapsule_SetContext()` is used here:

https://github.com/pybind/pybind11/blob/da104a9efd0357b5c144c67eb641ae0b9e23012d/include/pybind11/pytypes.h#L1846

But I'd expect that the rest of the world handles destructors via `PyCapsule_SetDestructor()` & `PyCapsule_GetDestructor()`. Could this lead to leaks, because we're never calling PyCapsule_GetDestructor() anywhere in pybind11?

I just looked, even Python 2.7 had `PyCapsule_SetDestructor()` & `PyCapsule_GetDestructor()` already.

Reply:
It turns out my worries here were unfounded. Based on feedback from @wjakob, I came to realize that my core misunderstanding was that `py::capsule` is controlling the lifetime (in a C++ dtor) but there is no C++ dtor here and the lifetime it is (of course) tied to the Python refcount of the capsule. This is connected to having misunderstood the `nullptr` for the 2nd argument of `PyCapsule_New`, that is the `name`, not the `destructor`.

I.e. everything is fine here. But having looked at this pretty closely now, I decided this minor cleanup will be useful: #4238